### PR TITLE
added api for add-rem-incr-decr

### PR DESCRIPTION
**Pull request by JatuFaique**
This PR has:
API integration for cart functionalities:
Add to cart
Remove from cart
Increment / Decrement cart

### CODE REVIEW

File: src/components/CartCard.js [hunk 1]
```diff
@@ -1,33 +1,85 @@
...
-    });
+  const handleQtyDecre = async (data) => {
+    try {
+      const encT = localStorage.getItem("token");
+      console.log(encT, data);
+      const res = await axios.post(
+        `/api/user/cart/${data._id}`,
+        {
+          action: {
+            type: "decrement",
```
Comment (sameep-s) [inline src/components/CartCard.js:17]: you can export actions as constants.

File: src/components/CartCard.js [hunk 2]
```diff
@@ -1,33 +1,85 @@
...
+          },
+        },
+        {
+          headers: {
+            authorization: encT,
+          },
+        }
+      );
+      cartDispatch({
+        type: "DECREASE_QTY",
```
Comment (sameep-s) [inline src/components/CartCard.js:27]: You can export these actions as constants.

File: src/reducers/cartReducer.js [hunk 3]
```diff
@@ -4,58 +4,76 @@ export const initialState = {
...
+      return {
+        ...state,
+        cart: action.payload,
+      };
+    case "REMOVE_FROM_CART":
+      return {
+        ...state,
+        cart: action.payload,
+      };
+    // case "ADD_TO_CART":
```
Comment (sameep-s) [inline src/reducers/cartReducer.js:40]: you can remove these comments

**Review by sameep-s — APPROVED** [review]
Code is good however, there is a lot of commented code you can remove that, and you can use constants for dispatch actions. 
These are just a few suggestions, otherwise code looks fine.

**Review by naazneen — APPROVED** [review]
Looks good to merge, Liked the implementations and handling of cart api status.

**Pull request merged**